src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs

```diff
@@ -56,12 +56,17 @@ public CSharpUseIndexOperatorDiagnosticAnalyzer()
 
         protected override void InitializeWorker(AnalysisContext context)
         {
-            context.RegisterCompilationStartAction(startContext =>
+            context.RegisterCompilationStartAction(context =>
```

Comment:
**Below callbacks were originally registered using the `AnalysisContext` rather than the `CompilationStartAnalysisContext`**

I'm not sure if this makes any different, but we should be using the most-nested context anyway, so shadowing is very intended.